main.tex

```diff
@@ -578,105 +732,156 @@ \subsubsection{Malicious Coordinator}
 
 \subsection{Unlinkability}
 
-A mixing scheme should ensure that any links between registered inputs and outputs are only known to their owners. The unlinkability property in~\cite{chase2019signal} ensures that presentation of credentials cannot be linked back to their issuance. Since every registration request is only associated with a single input or output, only information that would need be broadcast publicly on the blockchain if the protocol succeeds is revealed directly.
+A mixing scheme should ensure that any links between registered inputs and outputs are only known to their owners. The unlinkability property in~\cite{chase2019signal} ensures that the presentation of credentials cannot be linked back to their issuance. Since every registration request is only associated with a single input or output, the only information that would need to be broadcast publicly on the blockchain if the protocol succeeds is revealed directly.
 
-However, there are other means participants or the coordinator could reduce or potentially eliminate the achieved privacy guarantees.
+However, because the protocol may be repeated several times before resulting in a valid transaction, and because the transaction data is still overt the unlinkability of credentials is not enough to guarantee privacy, and there are other means participants or the coordinator could attack it.
 
 \subsubsection{Passive attacks}
 
-As in \cref{sec:examples} we can consider registration requests as vertices labeled by $\Delta_a$ of a graph where credentials are edges labeled by the amount from issuance to presentation. In the unified protocol this is a directed acyclic graph with indegree and outdegree $k$. The unlinkabilility of credentials and hiding property of the amount commitments obscure the edges and their labels. The serial number and balance proofs enforce a global invariant where the sums of the inwards and outwards edges of every vertex differ by its label.
+We can model registration requests as vertices of a directed acyclic graph labeled by $\Delta_a$ and credentials as edges labelled by the amount in the attribute, connecting the registration request where a credential was issued to where it was presented, much like the graph depicted in \cref{sec:examples}. Apart from the bootstrap requests which have indegree 0 (no credentials are presented) and final output requests which have outdegree 0 (requested credentials are not presented in any subsequent request), this graph is $k$-regular. The serial number and balance proofs enforce a global invariant where the sums of the inwards and outwards edges of every vertex differ by its label, and so long as honest participants are able to make their registrations it will be balanced (both vertex and edge labels will sum to 0).
```

Comment:
> (requested credentials are not presented in any subsequent request)

Why not? I think it could be possible to request credentials during output registration and present them in subsequent requests.

Reply:
this refers to the last credentials requested, since no registration follows, there is no subsequent request to present them in, it's really just there to make sure the following statement that the sub-graph of interest is k-regular is accurate, since the only way for a directed acyclic graph to truly be k-regular is if it's infinite...

i'm struggling to rephrase it right now, but i will try again a bit later if nobody else suggests anything, it can definitely be clarified

Reply:
is this enough?

```suggestion
We can model registration requests as vertices of a directed acyclic graph labeled by $\Delta_a$ and credentials as edges labelled by the amount in the attribute, connecting the registration request where a credential was issued to where it was presented, much like the graph depicted in \cref{sec:examples}. Apart from the bootstrap requests which have indegree 0 (no credentials are presented) and final output requests which have outdegree 0 (credentials must be requested but not used since there are no subsequent registrations), this graph is $k$-regular. The serial number and balance proofs enforce a global invariant where the sums of the inwards and outwards edges of every vertex differ by its label, and so long as honest participants are able to make their registrations it will be balanced (both vertex and edge labels will sum to 0).
```

Reply:
Yes.